### DISCUSSION TITLE
Apollo.primevue v4.4 - TypeScript migration

### DISCUSSION THREAD

**Discussion by dominiksulak** [primefaces/community#183] [PrimeVue Templates] [answered]
I have an Vue app using TS. I want to use apollo template but its written in JS. Is it ok to migrate apollo app template into TypeScript or its genereally not recommended? Also could you please provide me how to do the JS->TS migration? 

Thanks a lot!

**Comment by donPuerto** [comment]
Same here

**Comment by sezisfurkan** [comment, marked as answer]
Thank you for your purchase, Apollo is a vite project in JS, the TS version has the following format;

https://vite.new/vanilla-ts

1. Add config files to root folder.

- tsconfig.json
- tsconfig.node.json

I'd suggest turning linting part to false during migration of layout code to TS. Once it runs with TS, you can turn it on for further improvements.

at root folder.

2. The package.json file should include typescript and vue-tsc as a dev dependency;
```
"devDependencies": {
    "@vitejs/plugin-vue": "^4.2.3",
    "typescript": "^5.0.2",
    "vite": "^4.4.8",
    "vue-tsc": "^1.8.8"
  }
```
  
3.Rename main.js to main.ts

4.In App.vue, add lang="ts" to the script element.

5.Layout folder is the only necessary part, other files like views are demos. .vue files under src/layout should have lang="ts" as well. Now you can add types e.g.;


```
changeTopbarColor(event, topbarColor, logo) {
            this.$emit('topbar-color-change', topbarColor, logo);
            event.preventDefault();
        },


       changeTopbarColor(event:MouseEvent, topbarColor: string, logo: string) {
            this.$emit('topbar-color-change', topbarColor, logo);
            event.preventDefault();
        },
```


If you have any questions during migration, please get in touch with us.
